Sync/SyncTelemetryUtils.swift

```diff
@@ -468,6 +468,85 @@ public class GleanSyncOperationHelper {
         GleanMetrics.Pings.shared.tempSync.submit()
     }
 
+    public func reportTelemetry(_ result: MozillaAppServices.SyncResult) {
+        guard let json = result.telemetryJson,
+              let telemetry = try? RustSyncTelemetryPing.fromJSONString(
+                jsonObjectText: json
+              ) else {
+            return
+        }
```

Comment:
nit: preserving our guard style

```suggestion
        guard let json = result.telemetryJson,
              let telemetry = try? RustSyncTelemetryPing.fromJSONString(jsonObjectText: json)
        else { return }
```